### PR TITLE
[Merged by Bors] - chore(*): add 3 `fun_prop` attrs

### PR DESCRIPTION
**Pull request by urkud**
---

This PR used to be about change of auto fields in `Homeomorph`, later I reverted that part of the patch.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by urkud** [comment]
!bench

File: Mathlib/Topology/Homeomorph.lean [hunk 1]
```diff
@@ -40,9 +40,9 @@ variable {X : Type*} {Y : Type*} {Z : Type*}
 structure Homeomorph (X : Type*) (Y : Type*) [TopologicalSpace X] [TopologicalSpace Y]
     extends X ≃ Y where
   /-- The forward map of a homeomorphism is a continuous function. -/
-  continuous_toFun : Continuous toFun := by continuity
+  continuous_toFun : Continuous toFun := by dsimp; fun_prop
```
Comment (urkud) [inline Mathlib/Topology/Homeomorph.lean:43]: Looking at the bench, the increase in `dsimp` time is greater than the decrease in `aesop` time. I don't know what's the right choice here. Without `dsimp`, it looks at goals like `Continuous (Equiv.mk ..).invFun` and can't proceed.

**Comment by urkud** [comment]
!bench

**Comment by MichaelStollBayreuth** [comment]
<b>General information:</b>
```text
build:                                                 -8.3062 * 10⁹ (-0.00676%)
```


<b>1 file got slower by at least 10⁹ instructions:</b>
```text
Mathlib.Geometry.RingedSpace.PresheafedSpace.Gluing:   +1.4055 * 10⁹ (+0.347%)
```


<b>No file got slower by at least 10%.</b>


<b>5 files got faster by at least 10⁹ instructions:</b>
```text
Mathlib.Analysis.NormedSpace.PiTensorProduct.InjectiveSeminorm:
                                                       -3.5123 * 10⁹ (-1.21%)
Mathlib.Topology.Category.TopCat.Limits.Pullbacks:     -1.6817 * 10⁹ (-4.16%)
Mathlib.Topology.Algebra.Field:                        -1.5747 * 10⁹ (-13.5%)
Mathlib.Geometry.RingedSpace.OpenImmersion:            -1.5114 * 10⁹ (-0.804%)
Mathlib.CategoryTheory.Category.Cat.Limit:             -1.4023 * 10⁹ (-1.37%)
```


<b>1 files got faster by at least 10%:</b>
```text
Mathlib.Topology.Algebra.Field:                                       -13.5%
```

**Comment by grunweg** [comment]
See also #13994, which does this differently (your choice might be better!) and goes further.

**Comment by MichaelStollBayreuth** [comment]
#13994 gives a speed-up nearly ten times as large.

**Comment by j-loreaux** [comment]
As I mentioned in #13994, we should not be addin `fun_prop` as default values until #11092 is finished.

File: Mathlib/Topology/Homeomorph.lean [hunk 2]
```diff
@@ -40,9 +40,9 @@ variable {X : Type*} {Y : Type*} {Z : Type*}
 structure Homeomorph (X : Type*) (Y : Type*) [TopologicalSpace X] [TopologicalSpace Y]
     extends X ≃ Y where
   /-- The forward map of a homeomorphism is a continuous function. -/
-  continuous_toFun : Continuous toFun := by continuity
+  continuous_toFun : Continuous toFun := by dsimp only; fun_prop
```
Comment (urkud) [inline Mathlib/Topology/Homeomorph.lean:43]:
```suggestion
  continuous_toFun : Continuous toFun := by continuity
```

File: Mathlib/Topology/Homeomorph.lean [hunk 3]
```diff
@@ -40,9 +40,9 @@ variable {X : Type*} {Y : Type*} {Z : Type*}
...
   /-- The inverse map of a homeomorphism is a continuous function. -/
-  continuous_invFun : Continuous invFun := by continuity
+  continuous_invFun : Continuous invFun := by dsimp only; fun_prop
```
Comment (urkud) [inline Mathlib/Topology/Homeomorph.lean:45]:
```suggestion
  continuous_invFun : Continuous invFun := by continuity
```

**Comment by urkud** [comment]
@j-loreaux I reverted the `:= by ...` parts of this PR to avoid conflicts with @grunweg 's work. Now it's just +3 `@[fun_prop]` attrs.

**Comment by j-loreaux** [comment]
bors d+

wait for CI of course

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**